src/GraphQL/TypeExtensions.cs

```diff
@@ -179,6 +143,8 @@ public static Type GetGraphTypeFromType(this Type type, bool isNullable = false)
                 graphType = listType.MakeGenericType(elementType);
             }
 
+            graphType = GraphQL.GraphTypeRegistry.Get(type);
```

Comment:
I believe this needs to be moved above the `type.IsArray` check to fix the tests.